modules/ossm-extensions-ref-wasmplugin.adoc

```diff
@@ -0,0 +1,175 @@
+|string
+|Credentials to use for OCI image pulling. The name of a secret in the same namespace as the `WasmPlugin` that contains a pull secret which is to be used to authenticate against the registry when pulling the image.
+|No
+
+|spec.phase
+|PluginPhase
+|Determines where in the filter chain this `WasmPlugin` is injected.
+|No
+
+|spec.priority
+|`int64`
+|Determines ordering of `WasmPlugins` in the same `phase`. When multiple `WasmPlugins` are applied to the same workload in the same `phase`, they will be applied by priority, in descending order. If `priority` is not set, or two `WasmPlugins` exist with the same
+value, the ordering will be deterministically derived from name and namespace of the `WasmPlugins`. Defaults to `0`.
+|No
+
+|spec.pluginName:
+|string
+|The plugin name used in the Envoy configuration. Some .wasm modules might require this value to select the Wasm plugin to execute.
+|No
+
+|spec.pluginConfig
+|Struct
+|The configuration that will be passed on to the plugin.
+|No
+
+|spec.pluginConfig.verificationKey
+|string
+|Public key that will be used to verify signatures of signed OCI images or Wasm modules. Must be supplied in PEM format.
+|No
+|===
+
+`WorkloadSelector` specifies the criteria used to determine if a policy can be applied to a proxy. The matching criteria includes the metadata associated with a proxy, workload instance information such as labels attached to the pod/VM, or any other information that the proxy provides to Istio during the initial handshake. If multiple conditions are specified, all conditions need to match in order for the workload instance to be selected. Currently, only label based selection mechanism is supported.
+
+.WorkloadSelector
+[options="header"]
+[cols="a, a, a, a"]
+|===
+| Field | Type | Description | Required
+|matchLabels
+|map<string, string>
+|One or more labels that indicate a specific set of pods/VMs on which a policy should be applied. The scope of label search is restricted to the configuration namespace in which the resource is present.
+|Yes
+|===
+
+`PullPolicy` specifies the pull behavior to be applied when fetching an OCI image.
+
+.PullPolicy
+[options="header"]
+[cols="a, a"]
+|===
+| Field | Description
+|UNSPECIFIED_POLICY
+|Defaults to IfNotPresent, except for OCI images with tag latest, for which the default will be Always.
+
+|IfNotPresent
+|If an existing version of the image has been pulled before, that will be used. If no version of the image is present locally, we will pull the latest version.
+
+|Always
+|We will always pull the latest version of an image when applying this plugin.
+|===
+
+`Struct` represents a structured data value, consisting of fields which map to dynamically typed values. In some languages, Struct might be supported by a native representation. For example, in scripting languages like JS a struct is represented as an object. The details of that representation are described together with the proto support for the language.
```

Comment:
Replace `JS` with `JavaScript`

Reply:
Remove the final sentence `The details of that representation are described together with the proto support for the language.`